torch/csrc/jit/graph_executor.cpp

```diff
@@ -732,13 +733,60 @@ struct GraphExecutorImpl {
   std::mutex compile_mutex;
 };
 
-GraphExecutor::GraphExecutor(std::shared_ptr<Graph> graph, bool optimize)
-    : pImpl(new GraphExecutorImpl(std::move(graph), optimize)) {}
+struct ProfilingGraphExecutorImpl : public GraphExecutorImpl {
```

Comment:
This thing is going to grow a lot, it should be in a separate file.

Reply:
Also we should probably have a base class that both inherit from. Because as it is now, it is inheriting a lot of implementation from graph executor. I don't think we want that in the end, so this organization will get confusing.